### ISSUE TITLE
Warn if error message doesn't conform to style guideline

### ISSUE THREAD

**Issue by kangalioo** [rust-lang/rust-clippy#8172]
### What it does

According to the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/interoperability.html#error-types-are-meaningful-and-well-behaved-c-good-err):

> The error message given by the Display representation of an error type should be lowercase without trailing punctuation, and typically concise.

Clippy could make sure that error message are lowercase without trailing punctuation, by checking
- the `Display` impl on types that have an `Error` impl
- `.ok_or("error message")`, `.map_err(|_| "error message")` and similar methods

### Lint Name

misformatted_error_message

### Category

style

### Advantage

An error message that conforms to the official guidelines fits more easily into the ecosystem

### Example

```rust
.ok_or_else(|| format!("Unknown wifescore or grade `{}`!", threshold_str))?
```

Could be written as:

```rust
.ok_or_else(|| format!("unknown wifescore or grade `{}`", threshold_str))?
```

**Comment by kangalioo** [comment]
Just noticed this suggestion is indirectly included in #1798
